### ISSUE TITLE
`working-directory` is being ignored

### ISSUE THREAD

**Issue by Zerotask** [actions/upload-pages-artifact#74]
In my job, I have set a working-directory (docs/user) right at the start.
In this directory I run a build step which creates a `dist` directory (docs/user/dist).

If you try to run
```yaml
- name: Upload artifact
        uses: actions/upload-pages-artifact@v2
        with:
          path: "dist"
```

you will get the error:
```
Run actions/upload-pages-artifact@v2
  with:
    path: dist
    name: github-pages
    retention-days: 1
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.11.4/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.11.4/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.4/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.4/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.4/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.11.4/x64/lib
    GITHUB_PAGES: true
Run tar \
  tar \
    --dereference --hard-dereference \
    --directory "$INPUT_PATH" \
    -cvf "$RUNNER_TEMP/artifact.tar" \
    --exclude=.git \
    --exclude=.github \
    .
  shell: /usr/bin/sh -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.11.4/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.11.4/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.4/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.4/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.4/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.11.4/x64/lib
    GITHUB_PAGES: true
    INPUT_PATH: dist
tar: dist: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
Error: Process completed with exit code 2.
```

I'd assume that it considers my previously set `working-directory`, but I had to change it to `path: "docs/user/dist"`

**Comment by ashiishme** [comment]
Have you managed to fix it? I am experiencing same issue.

**Comment by Zerotask** [comment]
> Have you managed to fix it? I am experiencing same issue. 

Yes, as I wrote, you have to set the absolute path, not the relative path.

**Comment by ashiishme** [comment]
I have the absolute path defined for the uploading but it seems to fail with same error.

`docs/out` < this is the directory I am pointing to.

`docs` is where my application is located.

**Comment by ashiishme** [comment]
nvm, in my case it was a different issue but indeed the absolute path is the solution. The out directory was not generated in my case because of the missing property in my config file as per the new changes from NextJS but after fixing it, the absolute path fixed the upload issue. Thanks.

**Comment by arvinsim** [comment]
@ashiishme Seems that some tutorials do not reflect the new NextJS changes. Hence why we get this error.

**Comment by NeoSahadeo** [comment]
## Artifact error exit code: 2 when build Nextjs
Moved a bunch of files around between projects and that broke the build process

## Solution
Fixed NextJs build workflow by renaming the next.config.mjs to next.config.js and changing the export statement to use commonjs type.

[Build Error in question](https://github.com/NeoSahadeo/BigDSmallNBlog/actions/runs/7602344912/job/20702791687)

**Comment by shaqdeff** [comment]
> ## Artifact error exit code: 2 when build Nextjs
> Moved a bunch of files around between projects and that broke the build process
> 
> ## Solution
> Fixed NextJs build workflow by renaming the next.config.mjs to next.config.js and changing the export statement to use commonjs type.
> 
> [Build Error in question](https://github.com/NeoSahadeo/BigDSmallNBlog/actions/runs/7602344912/job/20702791687)

@NeoSahadeo My guy! Million thanks for this! I've stressing for hours with his till I came across your answer! Bless you!

**Comment by oyal** [comment]
Related to this: `actions/configure-pages@v4`. Please use v5 version.

```yml
- name: Setup Pages
  uses: actions/configure-pages@v5
  with:
    static_site_generator: next
``` 

https://github.com/actions/configure-pages/pull/137
